### DISCUSSION TITLE
Exiting impersonation logs out instead

### DISCUSSION THREAD

**Discussion by stephanvierkant** [symfony/symfony#45622] [Q&A]
When I exit the impersonation I got logged out instead. This is what I found in the logs:

> Cannot refresh token because user has changed.

Then I tried to find out why the user has changed, and `$this` in `\Symfony\Component\Security\Core\User\EquatableInterface::isEqualTo` gives `Proxies\__CG__\App\Entity\User` so the `username` is `null`.

I'm using Symfony 6. I don't understand why this happens. Any idea where to look?

**Comment by lyrixx** [comment]
Hello,

Could you create a [reproducer](https://symfony.com/doc/current/contributing/code/reproducer.html) please?

**Comment by simonsolutions** [comment]
Possible same issue as in PR https://github.com/symfony/symfony/pull/46748

Referencing to the issue https://github.com/symfony/symfony/issues/47356?!
